lib/configcat/rolloutevaluator.rb

```
@@ -1,197 +1,760 @@
+      if prerequisite_comparison_value_type != SettingType.to_type(prerequisite_flag_setting_type)
+        raise "Type mismatch between comparison value '#{prerequisite_comparison_value}' and prerequisite flag '#{prerequisite_key}'"
+      end
+
+      prerequisite_condition = "Flag '#{prerequisite_key}' #{PREREQUISITE_COMPARATOR_TEXTS[prerequisite_comparator]} '#{prerequisite_comparison_value}'"
+
+      # Circular dependency check
+      visited_keys = context.visited_keys
+      visited_keys.push(context.key)
+      if visited_keys.include?(prerequisite_key)
+        depending_flags = visited_keys.push(prerequisite_key).map { |s| "'#{s}'" }.join(' -> ')
+        raise "Circular dependency detected between the following depending flags: #{depending_flags}."
+      end
+
+      if log_builder
+        log_builder.append(prerequisite_condition)
+        log_builder.new_line('(').increase_indent
+        log_builder.new_line("Evaluating prerequisite flag '#{prerequisite_key}':")
+      end
+
+      prerequisite_value, _, _, _, _ = evaluate(key: prerequisite_key, user: context.user, default_value: nil, default_variation_id: nil,
+                                                config: config, log_builder: log_builder, visited_keys: context.visited_keys)
+
+      visited_keys.pop if visited_keys
+
+      if log_builder
+        log_builder.new_line("Prerequisite flag evaluation result: '#{prerequisite_value}'.")
+        log_builder.new_line("Condition (Flag '#{prerequisite_key}' #{PREREQUISITE_COMPARATOR_TEXTS[prerequisite_comparator]} '#{prerequisite_comparison_value}') evaluates to ")
+      end
+
+      if prerequisite_comparator == PrerequisiteComparator::EQUALS
+        prerequisite_condition_result = true if prerequisite_value == prerequisite_comparison_value
+      elsif prerequisite_comparator == PrerequisiteComparator::NOT_EQUALS
+        prerequisite_condition_result = true if prerequisite_value != prerequisite_comparison_value
+      end
```

We shouldn't let the evaluation continue if the comparator is invalid (neither EQUALS, nor NOT_EQUALS). Let's throw an exception like here: https://github.com/configcat/php-sdk/blob/60f4f5e14a1db3898834a22e44b30d850b7dc3c7/src/RolloutEvaluator.php#L995

This also applies to `evaluate_segment_condition` and `evaluate_user_condition`.

(Unfortunately, this issue is also present in the Python SDK. Please correct that as well.)